internal/github/github.go

```diff
@@ -117,6 +129,15 @@ func (c GHClient) IsPRMergedOrClosed(ctx context.Context, owner, repo string, pr
 	return sha, pr.GetState() == "closed", mergedAt, nil
 }
 
+func (c GHClient) GetPRHeadSHA(ctx context.Context, owner, repo string, prNumber int) (string, error) {
+	pr, _, err := c.client.PullRequests.Get(ctx, owner, repo, prNumber)
+	if err != nil {
+		return "", fmt.Errorf("failed to query GitHub: %w", err)
```

Comment:
Adds slightly more detail; disregard if error would provide that info or or in some other way is redundant.
```suggestion
		return "", fmt.Errorf("failed to query GitHub for PR HEAD SHA: %w", err)
```

Reply:
Good idea, I have committed it 👍